doc/api/util.md

```diff
@@ -398,6 +398,107 @@ util.inspect.defaultOptions.maxArrayLength = null;
 console.log(arr); // logs the full array
 ```
 
+## util.promisify(original)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `original` {Function}
+
+Takes a function following the common Node.js callback style, i.e. taking a
+`(err, value) => ...` callback as the last argument, and returns a version
+that returns promises.
+
+For example:
+
+```js
+const util = require('util');
+const fs = require('fs');
+
+const stat = util.promisify(fs.stat);
+stat('.').then((stats) => {
+  // Do something with `stats`
+}).catch((error) => {
+  // Handle the error.
+});
+```
+
+Or, equivalently using `async function`s:
+
+```js
+const util = require('util');
+const fs = require('fs');
+
+const stat = util.promisify(fs.stat);
+
+async function callStat() {
+  const stats = await stat('.');
+  console.log(`This directory is owned by ${stats.uid}`);
+}
+```
+
+If there is an `original[util.promisify.custom]` property present, `promisify`
+will return its value, see [Custom promisified functions][].
+
+If `original` calls its callback argument with more than 2 parameters, a
+Promise for an Array of those parameters is returned instead, for example:
```

Comment:
I think the implicit behavior switch here might be confusing. Bluebird originally had this behavior in its `promisify` function, but it was [changed later in a major release](https://github.com/petkaantonov/bluebird/issues/307) because it caused compatibility issues when an additional argument was added to a callback. The [3.0 promisify function](http://bluebirdjs.com/docs/api/promise.promisify.html) has an explicit option for converting to an array.

I think it would be better to always use the first value in the callback (possibly with an opt-in for returning an array, like Bluebird has) rather than sometimes returning an array and sometimes returning a single value.